add supported chains to chain fusion section. (#3122)
dfinity/portal: docs/developer-docs/multi-chain/supported-chains.mdx

---
keywords:
  [
    advanced,
    Chain Fusion,
    Bitcoin,
    Ethereum,
    Arbitrum,
    Optimism,
    Polygon,
    Blast,
    Base,
    Solana,
    XRP,
    Toncoin,
    Dogecoin,
    Cardano,
    TRON,
    Avalanche,
    Polkadot,
    Cosmos,
    NEAR,
    Kaspa,
    Aptos,
    Monero,
    Hedera,
    Filecoin,
    Stacks,
    Sui,
    Arweave,
    Stellar,
    Chain Fusion,
  ]
---

import useBaseUrl from "@docusaurus/useBaseUrl";
import { MarkdownChipRow } from "/src/components/Chip/MarkdownChipRow";
import { AlphaChip } from "/src/components/Chip/AlphaChip";
import { BetaChip } from "/src/components/Chip/BetaChip";

# Supported chains

<MarkdownChipRow labels={["Advanced", "Chain Fusion"]} />

## Overview

> A chain can be supported if it leverages an authentication method that is
> supported by the threshold signing protocols of the Internet Computer, and has
> reliable RPC providers supporting IPv6. For some chains, DFINITY or the
> community has built libraries or canisters to simplify the integration.

Reviewer: ```suggestion
Other blockchain networks can be integrated with ICP if they leverage an authentication method that is
supported by the threshold signing protocols of the Internet Computer, and has
reliable RPC providers supporting IPv6. 

For some chains, DFINITY or the ICP community has built libraries or canisters to simplify the integration.
```

> ## Selection of supported chains

Reviewer: ```suggestion
## Supported chains
```

> The following table lists a selection of supported chains and their respective
> authentication methods, available integrations, and chain-key tokens. EVM
> compatible chains other than Ethereum are not listed explicitly, but they are
> supported in the same way as Ethereum.

Reviewer: ```suggestion
The following table lists a selection of supported chains and their respective
authentication methods, available integrations, and chain-key tokens. 

:::caution
EVM-compatible chains other than Ethereum are not listed explicitly, but they are
supported in the same way as Ethereum through the EVM RPC canister.
:::
```

> - **Authentication**: Authentication methods, i.e. signature schemes used to
>   authenticate transactions, that are supported by the Internet Computer's
>   threshold signing capabilities.

Reviewer: ```suggestion
- **Authentication**: Authentication methods, such as signature schemes used to authenticate transactions, that are supported by ICP's threshold signing capabilities.
```

> - **Available Integrations**: The available integrations for each chain, i.e.
>   the methods by which the Internet Computer can interact with the chain.

Reviewer: ```suggestion
- **Available integrations**: The available integrations for each chain, i.e.
  the methods by which ICP can interact with the chain.
```

| Filecoin  | ECDSA                                                                  | [RPC via HTTPS Outcalls](docs/developer-docs/smart-contracts/advanced-features/https-outcalls/https-outcalls-overview.mdx)                     |
| Stacks    | ECDSA                                                                  | [RPC via HTTPS Outcalls](docs/developer-docs/smart-contracts/advanced-features/https-outcalls/https-outcalls-overview.mdx)                     |
| Sui       | ECDSA, EdDSA                                                           | [RPC via HTTPS Outcalls](docs/developer-docs/smart-contracts/advanced-features/https-outcalls/https-outcalls-overview.mdx)                     |
| Arweave   | [Not yet supported](https://github.com/ArweaveTeam/arweave/issues/486) | [RPC via HTTPS Outcalls](docs/developer-docs/smart-contracts/advanced-features/https-outcalls/https-outcalls-overview.mdx)                     |
| Stellar   | EdDSA                                                                  | [RPC via HTTPS Outcalls](docs/developer-docs/smart-contracts/advanced-features/https-outcalls/https-outcalls-overview.mdx)                     |

> :::note

Reviewer: ```suggestion
:::caution
```

Threshold Schnorr and EdDSA signing are not available in production yet.
Threshold Schnorr signing will be released with the Deuterium milestone
scheduled for August 2024. Threshold EdDSA signing will be released with the
Helium milestone later this year. However, both signing schemes are already
available as a [developer preview](https://github.com/domwoe/schnorr_canister).

:::

> ## The chain you want to integrate is not listed?

Reviewer: ```suggestion
### Looking for a chain that is not listed?
```

> Most likely, the chain you want to integrate is still supported by the Internet
> Computer. Check the docs of the chain you want to integrate to find out which
> authentication/signature schemes are supported.

Reviewer: ```suggestion
Most likely, the chain you want to integrate is still supported by ICP. Check the docs of the chain you want to integrate to find out which authentication methods or signature schemes are supported.
```

---
keywords:
  [
    advanced,
    Chain Fusion,
    Bitcoin,
    Ethereum,
    Arbitrum,
    Optimism,
    Polygon,
    Blast,
    Base,
    Solana,
    XRP,
    Toncoin,
    Dogecoin,
    Cardano,
    TRON,
    Avalanche,
    Polkadot,
    Cosmos,
    NEAR,
    Kaspa,
    Aptos,
    Monero,
    Hedera,
    Filecoin,
    Stacks,
    Sui,
    Arweave,
    Stellar,

>     Chain Fusion,

Reviewer: ```suggestion

```

:::

- **Authentication**: Authentication methods, such as signature schemes used to
  authenticate transactions, that are supported by ICP's threshold signing
  capabilities.

- **Available integrations**: The available integrations for each chain, i.e.
  the methods by which ICP can interact with the chain.

- **Assets**: Assets build on top of the integrations, such as chain-key tokens.

| Chain     | Authentication                                                         | Available Integrations                                                                                                                         | Assets                                                                                                                                          |
| --------- | ---------------------------------------------------------------------- | ---------------------------------------------------------------------------------------------------------------------------------------------- | ----------------------------------------------------------------------------------------------------------------------------------------------- |

> | Bitcoin   | ECDSA, Schnorr                                                         | Direct                                                                                                                                         | [ckBTC](docs/developer-docs/multi-chain/chain-key-tokens/ckbtc/overview.mdx)                                                                    |
> | Ethereum  | ECDSA                                                                  | [EVM RPC](docs/developer-docs/multi-chain/ethereum/evm-rpc/overview.mdx), [ic-web3-rs](https://github.com/horizonx-tech/ic-web3-rs)            | [ckETH](docs/developer-docs/multi-chain/chain-key-tokens/cketh/overview.mdx), ckERC20                                                           |
> | Solana    | EdDSA                                                                  | [Solana RPC](https://github.com/mfactory-lab/ic-solana) <span style={{ display: 'inline-block', verticalAlign: 'super' }}><AlphaChip /></span> | [gSOL](https://galactic-bridge-tzbjl.ondigitalocean.app/) <span style={{ display: 'inline-block', verticalAlign: 'super' }}><BetaChip /></span> |
> | XRP       | ECDSA, EdDSA                                                           | [RPC via HTTPS Outcalls](docs/developer-docs/smart-contracts/advanced-features/https-outcalls/https-outcalls-overview.mdx)                     |                                                                                                                                                 |
> | Toncoin   | EdDSA                                                                  | [RPC via HTTPS Outcalls](docs/developer-docs/smart-contracts/advanced-features/https-outcalls/https-outcalls-overview.mdx)                     |                                                                                                                                                 |
> | Dogecoin  | ECDSA                                                                  | [RPC via HTTPS Outcalls](docs/developer-docs/smart-contracts/advanced-features/https-outcalls/https-outcalls-overview.mdx)                     |                                                                                                                                                 |
> | Cardano   | ECDSA, EdDSA, Schnorr                                                  | [RPC via HTTPS Outcalls](docs/developer-docs/smart-contracts/advanced-features/https-outcalls/https-outcalls-overview.mdx)                     |                                                                                                                                                 |
> | TRON      | ECDSA                                                                  | [RPC via HTTPS Outcalls](docs/developer-docs/smart-contracts/advanced-features/https-outcalls/https-outcalls-overview.mdx)                     |                                                                                                                                                 |
> | Avalanche | ECDSA                                                                  | [RPC via HTTPS Outcalls](docs/developer-docs/smart-contracts/advanced-features/https-outcalls/https-outcalls-overview.mdx)                     |
> | Polkadot  | ECDSA, EdDSA                                                           | [RPC via HTTPS Outcalls](docs/developer-docs/smart-contracts/advanced-features/https-outcalls/https-outcalls-overview.mdx)                     |
> | Cosmos    | ECDSA, EdDSA                                                           | [RPC via HTTPS Outcalls](docs/developer-docs/smart-contracts/advanced-features/https-outcalls/https-outcalls-overview.mdx)                     |
> | NEAR      | EdDSA                                                                  | [RPC via HTTPS Outcalls](docs/developer-docs/smart-contracts/advanced-features/https-outcalls/https-outcalls-overview.mdx)                     |
> | Kaspa     | ECDSA                                                                  | [RPC via HTTPS Outcalls](docs/developer-docs/smart-contracts/advanced-features/https-outcalls/https-outcalls-overview.mdx)                     |
> | Aptos     | ECDSA, EdDSA                                                           | [RPC via HTTPS Outcalls](docs/developer-docs/smart-contracts/advanced-features/https-outcalls/https-outcalls-overview.mdx)                     |
> | Monero    | EdDSA                                                                  | [RPC via HTTPS Outcalls](docs/developer-docs/smart-contracts/advanced-features/https-outcalls/https-outcalls-overview.mdx)                     |
> | Hedera    | ECDSA, EdDSA                                                           | [RPC via HTTPS Outcalls](docs/developer-docs/smart-contracts/advanced-features/https-outcalls/https-outcalls-overview.mdx)                     |
> | Filecoin  | ECDSA                                                                  | [RPC via HTTPS Outcalls](docs/developer-docs/smart-contracts/advanced-features/https-outcalls/https-outcalls-overview.mdx)                     |
> | Stacks    | ECDSA                                                                  | [RPC via HTTPS Outcalls](docs/developer-docs/smart-contracts/advanced-features/https-outcalls/https-outcalls-overview.mdx)                     |
> | Sui       | ECDSA, EdDSA                                                           | [RPC via HTTPS Outcalls](docs/developer-docs/smart-contracts/advanced-features/https-outcalls/https-outcalls-overview.mdx)                     |
> | Arweave   | [Not yet supported](https://github.com/ArweaveTeam/arweave/issues/486) | [RPC via HTTPS Outcalls](docs/developer-docs/smart-contracts/advanced-features/https-outcalls/https-outcalls-overview.mdx)                     |
> | Stellar   | EdDSA                                                                  | [RPC via HTTPS Outcalls](docs/developer-docs/smart-contracts/advanced-features/https-outcalls/https-outcalls-overview.mdx)                     |

Reviewer: ```suggestion

| Aptos     | [ECDSA, EdDSA](https://aptos.dev/en/network/blockchain/accounts)                                                           | [RPC via HTTPS Outcalls](docs/developer-docs/smart-contracts/advanced-features/https-outcalls/https-outcalls-overview.mdx)                     |    
| Arweave   | [Not yet supported](https://github.com/ArweaveTeam/arweave/issues/486) | [RPC via HTTPS Outcalls](docs/developer-docs/smart-contracts/advanced-features/https-outcalls/https-outcalls-overview.mdx)                     |
| Avalanche | [ECDSA](https://docs.avax.network/reference/standards/cryptographic-primitives)                                                                 | [RPC via HTTPS Outcalls](docs/developer-docs/smart-contracts/advanced-features/https-outcalls/https-outcalls-overview.mdx)                     |
| Bitcoin   | [ECDSA, Schnorr](/docs/current/developer-docs/multi-chain/bitcoin/using-btc/generate-addresses)                                                        | Direct                                                                                                                                         | [ckBTC](docs/developer-docs/multi-chain/chain-key-tokens/ckbtc/overview.mdx)                                                                    |
| Cardano   | [ECDSA, EdDSA, Schnorr ](https://developers.cardano.org/docs/governance/cardano-improvement-proposals/cip-0049/)                                                 | [RPC via HTTPS Outcalls](docs/developer-docs/smart-contracts/advanced-features/https-outcalls/https-outcalls-overview.mdx)                     |                                                                                                                                                 |
| Cosmos    | [ECDSA, EdDSA](https://docs.cosmos.network/v0.50/learn/beginner/accounts)                                                           | [RPC via HTTPS Outcalls](docs/developer-docs/smart-contracts/advanced-features/https-outcalls/https-outcalls-overview.mdx)                     |
| Dogecoin  | ECDSA                                                                  | [RPC via HTTPS Outcalls](docs/developer-docs/smart-contracts/advanced-features/https-outcalls/https-outcalls-overview.mdx)                     |                                                                                                                                                 |
| Ethereum  | [ECDSA](/docs/current/developer-docs/multi-chain/ethereum/using-eth/generating-addresses)                                                                  | [EVM RPC](docs/developer-docs/multi-chain/ethereum/evm-rpc/overview.mdx), [ic-web3-rs](https://github.com/horizonx-tech/ic-web3-rs)            | [ckETH](docs/developer-docs/multi-chain/chain-key-tokens/cketh/overview.mdx), ckERC20                                                           |
| Filecoin  | [ECDSA](https://spec.filecoin.io/algorithms/crypto/signatures/)                                                                  | [RPC via HTTPS Outcalls](docs/developer-docs/smart-contracts/advanced-features/https-outcalls/https-outcalls-overview.mdx)                     |
| Hedera    | [ECDSA, EdDSA](https://docs.hedera.com/hedera/core-concepts/keys-and-signatures)                                                           | [RPC via HTTPS Outcalls](docs/developer-docs/smart-contracts/advanced-features/https-outcalls/https-outcalls-overview.mdx)                     |
| Kaspa     | [ECDSA](https://github.com/kaspanet/go-secp256k1/blob/master/ecdsa_privkey.go)                                                                  | [RPC via HTTPS Outcalls](docs/developer-docs/smart-contracts/advanced-features/https-outcalls/https-outcalls-overview.mdx)                     |
| Monero    | [EdDSA](https://getmonero.dev/cryptography/asymmetric/edwards25519.html)                                                                  | [RPC via HTTPS Outcalls](docs/developer-docs/smart-contracts/advanced-features/https-outcalls/https-outcalls-overview.mdx)                     |
| NEAR      | EdDSA                                                                  | [RPC via HTTPS Outcalls](docs/developer-docs/smart-contracts/advanced-features/https-outcalls/https-outcalls-overview.mdx)                     |
| Polkadot  | [ECDSA, EdDSA](https://wiki.polkadot.network/docs/learn-cryptography)                                                           | [RPC via HTTPS Outcalls](docs/developer-docs/smart-contracts/advanced-features/https-outcalls/https-outcalls-overview.mdx)                     |
| Solana    | [EdDSA](https://solana.com/docs/core/accounts#account)                                                                  | [Solana RPC](https://github.com/mfactory-lab/ic-solana) <span style={{ display: 'inline-block', verticalAlign: 'super' }}><AlphaChip /></span> | [gSOL](https://galactic-bridge-tzbjl.ondigitalocean.app/) <span style={{ display: 'inline-block', verticalAlign: 'super' }}><BetaChip /></span> |
| Stacks    | [ECDSA](https://github.com/stacks-network/keylib-py)                                                                  | [RPC via HTTPS Outcalls](docs/developer-docs/smart-contracts/advanced-features/https-outcalls/https-outcalls-overview.mdx)                     |
| Stellar   | [EdDSA](https://developers.stellar.org/docs/smart-contracts/smart-wallets#secp256r1)                                                                  | [RPC via HTTPS Outcalls](docs/developer-docs/smart-contracts/advanced-features/https-outcalls/https-outcalls-overview.mdx)                     |
| Sui       | [ECDSA, EdDSA](https://blog.sui.io/wallet-cryptography-specifications/)                                                           | [RPC via HTTPS Outcalls](docs/developer-docs/smart-contracts/advanced-features/https-outcalls/https-outcalls-overview.mdx)                     |
| XRP       | [ECDSA, EdDSA](https://xrpl.org/docs/concepts/accounts/cryptographic-keys/#signing-algorithms)                                                           | [RPC via HTTPS Outcalls](docs/developer-docs/smart-contracts/advanced-features/https-outcalls/https-outcalls-overview.mdx)                     |                                                                                                                                                 |
| Toncoin   | EdDSA                                                                  | [RPC via HTTPS Outcalls](docs/developer-docs/smart-contracts/advanced-features/https-outcalls/https-outcalls-overview.mdx)                     |                                                                                                                                                 |
| TRON      | [ECDSA](https://github.com/tronprotocol/tips/issues/120)                                                                  | [RPC via HTTPS Outcalls](docs/developer-docs/smart-contracts/advanced-features/https-outcalls/https-outcalls-overview.mdx)                     |                                                                                                                                                 |
```

Reviewer: Reorganized by alphabetical order and added links to each chain's documentation that specifies its supported signatures

Author: I'm not sure if this is a good idea as links may change, and there's not a huge value in having these links available.
